docs: update outdated docs for `YN0019` (#3288)
yarnpkg/berry: packages/gatsby/content/advanced/error-codes.md

> A file from the cache has been detected unused by `yarn cache clean`.

Reviewer: `yarn cache clean` doesn't detect unused files anymore, it happens in the fetch step which is executed by any command that runs an install.

```suggestion
A file from the cache has been detected unused when installing dependencies.
```
